spec/jobs/send_weekly_alert_email_job_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe SendWeeklyAlertEmailJob, type: :job do
+  include ActiveJob::TestHelper
+
+  subject(:job) { described_class.perform_later }
+
+  let(:search_criteria) do
+    {
+      subject: 'English',
+      working_patterns: %w[full_time],
+      phases: %w[primary secondary]
+    }.to_json
+  end
+
+  let!(:subscription) { create(:subscription, search_criteria: search_criteria, frequency: :weekly) }
+  let!(:vacancies) { create_list(:vacancy, 5, :published_slugged) }
+
+  let(:mail) { double(:mail) }
+
+  it 'queues the job' do
+    expect { job }.to change(ActiveJob::Base.queue_adapter.enqueued_jobs, :size).by(1)
+  end
+
+  it 'is in the queue_daily_alerts queue' do
+    expect(job.queue_name).to eq('queue_weekly_alerts')
+  end
+
+  context 'with vacancies' do
+    before do
+      allow_any_instance_of(described_class).to receive(:vacancies_for_subscription) { vacancies }
+      allow(Rails).to receive(:env).and_return(ActiveSupport::StringInquirer.new('production'))
+    end
+
+    it 'sends an email' do
+      expect(Subscription).to receive(:weekly) { Subscription.where(frequency: :weekly) }
```

Comment:
Why do we need to specify this? Does the FactoryBot subscription not respond to `weekly` like the real subscription?

Reply:
Ideally we should just do `expect(Subscription).to receive(:weekly)` but for some reasons, when the code is executed `Subscription.weekly` returns nil. So I had to explicitly set what it should return in the spec